plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java

```diff
@@ -318,15 +318,26 @@ private void warnExtraConfig(Path extraContentDir) {
         }
     }
 
-    private Path getDeploymentContent() {
+    private Path getDeploymentContent() throws MojoExecutionException {
         final PackageType packageType = PackageType.resolve(project);
         final String filename;
         if (this.filename == null) {
             filename = String.format("%s.%s", project.getBuild().getFinalName(), packageType.getFileExtension());
         } else {
             filename = this.filename;
         }
-        return targetDir.toPath().resolve(filename);
+        Path deployment = targetDir.toPath().resolve(filename);
+        if (Files.notExists(deployment)) {
+            if (this.filename != null ) {
+                throw new MojoExecutionException("No deployment found wih name " + this.filename);
+            }
+            if (this.runtimeName != null ) {
+                throw new MojoExecutionException("No deployment found although a runtime-name has been set. "
```

Comment:
At first sight It was confusing to me why `runtime-name` is mentioned. I assume it is because when it is set, that mean user want match some deployment. So my understanding of error message is something like this

```
"No deployment " + filename +"found. As runtime-name has been set, that indicate that there is expected some deployment. A custom path to deployment can be set with <filename> parameter."
```